### PR TITLE
RR-441 populate UI with data and remove temporary code

### PR DESCRIPTION
**Pull request by chrimesdev**
## Description

Populate the UI with the data for the short route answers

https://dsdmoj.atlassian.net/browse/RR-441

### Screenshots

`A8335DY` 

```
{
  "problemRetrievingData": false,
  "inductionQuestionSet": "SHORT_QUESTION_SET",
  "data": {
    "updatedBy": "KROONEY_GEN",
    "updatedAt": "2023-10-19T10:39:30.080Z",
    "shortQuestionSetAnswers": {
      "additionalTraining": [
        "HGV_LICENCE"
      ],
      "otherAdditionalTraining": "",
      "educationalQualifications": []
    }
  }
}
```

Screenshot 2023-10-26 at 15 14 46

`A8408DY`

```
{
  "problemRetrievingData": false,
  "inductionQuestionSet": "SHORT_QUESTION_SET",
  "data": {
    "updatedBy": "KROONEY_GEN",
    "updatedAt": "2023-10-19T10:39:30.080Z",
    "shortQuestionSetAnswers": {
      "additionalTraining": [
        "HGV_LICENCE"
      ],
      "otherAdditionalTraining": "",
      "educationalQualifications": []
    }
  }
}
```

Screenshot 2023-10-26 at 15 34 17

### CODE REVIEW

File: server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk [hunk 1]
```diff
@@ -53,11 +48,13 @@ as the follow up questions are different.
           Other training or qualifications
         </dt>
         <dd class="govuk-summary-list__value">
-          {# TODO: Populate values and handle the for each statement #}
           <ul class="govuk-list">
-            {# for each qualification recorded #}
-            <li>[Qualification value]</li>
-            {# endfor #}
+            {% for additionalTraining in educationAndTraining.data.shortQuestionSetAnswers.additionalTraining %}
```
Comment (nathanrussell-moj-digital) [inline server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk:52]: I think this `for` and `if` are slightly wrong. I think you want to do it like I've done it here (for the long question set - you can replace `longQuestionSet` with `shortQuestionSet` and it should work):
```
          <ul class="govuk-list">
            {% for additionalTraining in educationAndTraining.data.longQuestionSetAnswers.additionalTraining %}
              {% if additionalTraining == 'OTHER' %}
                <li>{{ educationAndTraining.data.longQuestionSetAnswers.otherAdditionalTraining }}</li>
              {% else %}
                <li>{{ additionalTraining | formatAdditionalTraining }}</li>
              {% endif %}  
            {% endfor %}
          </ul>
```

File: server/filters/formatQualificationLevelFilter.ts [hunk 2]
```diff
@@ -0,0 +1,16 @@
+export default function formatQualificationLevelFilter(value: string): string {
+  const levelValue = LevelValues[value as keyof typeof LevelValues]
+  return levelValue
+}
+
+enum LevelValues {
```
Comment (nathanrussell-moj-digital) [inline server/filters/formatQualificationLevelFilter.ts:6]: I'm not 100% sure this is right, though I can see where it is in Gary's UI codebase so can understand why you've done it the way you have.
The problem is, the API returns us qualification levels that look like this:
```
    AchievedQualification: {
      /** @description This is the subject the inmate has chosen. */
      subject?: string
      /** @description This is the grade on the subject the inmate has chosen. */
      grade?: string
      /**
       * @description This is the level of  the subject the inmate has chosen.
       * @enum {string}
       */
      level?:
        | 'ENTRY_LEVEL_2'
        | 'ENTRY_LEVEL_3'
        | 'LEVEL_1'
        | 'LEVEL_2'
        | 'LEVEL_3'
        | 'LEVEL_4'
        | 'LEVEL_5'
        | 'LEVEL_6'
        | 'LEVEL_7'
        | 'LEVEL_8'
    }
```
IE. We get `ENTRY_LEVEL_2` and `ENTRY_LEVEL_3`; but no `ENTRY_LEVEL` ....

I'll talk to Gary and Sacintha about this apparent disconnect in their UI and API ... leave this one with me and I'll let you know what I find ...

Comment (nathanrussell-moj-digital) [reply server/filters/formatQualificationLevelFilter.ts:6]: OK, I think what you've done is correct, but there is an API bug that will prevent this working in all cases!
https://mojdt.slack.com/archives/C056PJR1T2A/p1698333864249309

Dont worry about changing your version here, its correct. The CIAG dev team can fix the mismatch in their API (which is where the problem is)

File: server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk (hunk 1 above)
Comment (chrimesdev) [reply server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk:52]: Yeah that makes more sense!

Comment (chrimesdev) [reply server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk:48]: Fixed

File: server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk [hunk 3]
```diff
@@ -10,41 +10,36 @@ as the follow up questions are different.
...
-      {# else if no qualifications recorded #}
-        <a class="govuk-link app-summary-card__change-link__link" href="{{ ciagInductionUrl }}/plan/create/{{ prisonerSummary.prisonNumber }}/qualifications-list/edit">Change<span class="govuk-visually-hidden">  whether they want educational qualifications recorded</span></a>
-      {# endif #}
+      {% if educationAndTraining.data.shortQuestionSetAnswers.educationalQualifications | length %}
+        <a class="govuk-link app-summary-card__change-link__link" href="{{ ciagInductionUrl }}/plan/create/{{ prisonerSummary.prisonNumber }}/qualifications-list/update">Change<span class="govuk-visually-hidden"> qualifications</span></a>
+      {% else %}
+        <a class="govuk-link app-summary-card__change-link__link" href="{{ ciagInductionUrl }}/plan/create/{{ prisonerSummary.prisonNumber }}/wants-to-add-qualifications/update">Change<span class="govuk-visually-hidden">  whether they want educational qualifications recorded</span></a>
```
[the remark this answers is not in the document]
Comment (chrimesdev) [reply server/views/pages/overview/partials/educationAndTrainingTab/_inductionShortQuestionSet.njk]: Fixed

**Review by nathanrussell-moj-digital — APPROVED** [review]
LGTM 👍

**Pull request merged**